### PR TITLE
Enable various error-prone checks

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

**Review by losipiuk — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PruneUnreferencedOutputs.java [hunk 1]
```diff
@@ -295,10 +295,10 @@ public PlanNode visitSpatialJoin(SpatialJoinNode node, RewriteContext<Set<Symbol
                     .build();
 
             ImmutableSet.Builder<Symbol> leftInputs = ImmutableSet.builder();
-            node.getLeftPartitionSymbol().map(leftInputs::add);
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PruneUnreferencedOutputs.java:298]: 🤯

File: pom.xml [hunk 2]
```diff
@@ -1798,6 +1798,7 @@
                                     -Xep:ObjectToString:ERROR
                                     -Xep:StaticQualifiedUsingExpression:ERROR
                                     -Xep:UnusedVariable:ERROR
+                                    -Xep:OptionalMapUnusedValue:ERROR
```
Comment (findepi) [inline pom.xml:1801]: keep sorted alphabetically

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java [hunk 3]
```diff
@@ -295,6 +295,7 @@ public void initialize(URI uri, Configuration conf)
     }
 
     @Override
+    @SuppressWarnings("UnnecessaryMethodReference")
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java:298]: what is the ignored problem and why you considered it the right thing to do?

at least add a comment, but this can indicate  a bigger problem

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveWriterFactory.java [hunk 4]
```diff
@@ -380,58 +380,53 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
...
-                                        "Inserting rows into such partition is not supported. " +
-                                        "The column '%s' in table '%s' is declared as type '%s', " +
-                                        "but partition '%s' declared column '%s' as type '%s'.",
-                                tableColumns.get(i).getName(),
-                                tableName,
-                                tableType,
-                                partitionName,
-                                existingPartitionColumns.get(i).getName(),
-                                partitionType));
+                case APPEND:
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveWriterFactory.java:385]: the comment on prev line should be indented

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HivePrincipal.java [hunk 5]
```diff
@@ -73,16 +73,14 @@ public HivePrincipal(@JsonProperty("type") PrincipalType type, @JsonProperty("na
     {
         this.type = requireNonNull(type, "type is null");
         requireNonNull(name, "name is null");
-        if (type == PrincipalType.USER) {
-            // In Hive user names are case sensitive
-            this.name = name;
-        }
-        else if (type == PrincipalType.ROLE) {
-            // In Hive role names are case insensitive
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HivePrincipal.java:81]: this comment is important but got lost

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HivePrincipal.java [hunk 6]
```diff
@@ -73,16 +73,14 @@ public HivePrincipal(@JsonProperty("type") PrincipalType type, @JsonProperty("na
...
-        }
-        else {
-            throw new IllegalArgumentException("Unsupported type: " + type);
+        switch (type) {
+            case USER:
+                this.name = name;
+                break;
+            case ROLE:
+                this.name = name.toLowerCase(ENGLISH);
+            default:
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HivePrincipal.java:82]: missing break

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java [hunk 7]
```diff
@@ -1179,6 +1179,7 @@ protected void doTestMismatchSchemaTable(
         }
     }
 
+    @SuppressWarnings("UnnecessaryOptionalGet")
```
Comment (findepi) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java:1182]: same here -- is the errorprone wrong?
at least add a comment, but this may indicate a bigger problem

**Review by findepi — CHANGES_REQUESTED** [review]

**Comment by findepi** [comment]
i'd suggest splitting PRs.  Such refactors require focus to review.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java (hunk 3 above)
Comment (wendigo) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java:298]: you can't pass super as a method reference (even if it satisfies functional interface here)

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java (hunk 7 above)
Comment (wendigo) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java:1182]: In this case it's wrong (false positive). I'll leave a comment here

**Comment by wendigo** [comment]
@findepi how should I split this review? I've done so by a check that is being enabled.

**Comment by wendigo** [comment]
AC @findepi

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java (hunk 3 above)
Comment (findepi) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java:298]: like here? 
https://github.com/trinodb/trino/blob/57da5d653d3dd91afba0e057ac6e703752f09078/plugin/trino-phoenix/src/test/java/io/trino/plugin/phoenix/TestPhoenixDistributedQueries.java#L133

or you mean something else?

**Comment by findepi** [comment]
> @findepi how should I split this review? I've done so by a check that is being enabled.

i meant separate PRs so that one can press the green button and move on

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java (hunk 7 above)
Comment (findepi) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java:1182]: it's an obvious but in the check.
it realized you're calling `xx.getDiscretePredicates().ifPresent()` and then `xx.getDiscretePredicates().get()`, but did NOT realize the `xx` part is not the same

please report to them and put link in the comment

**Comment by wendigo** [comment]
@findepi Got that. I'll do it for next checks

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java (hunk 7 above)
Comment (wendigo) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java:1182]: https://github.com/google/error-prone/issues/2101

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java (hunk 3 above)
Comment (wendigo) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java:298]: https://github.com/google/error-prone/issues/2102

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java [hunk 8]
```diff
@@ -295,10 +295,12 @@ public void initialize(URI uri, Configuration conf)
...
     public void close()
             throws IOException
     {
         try (Closer closer = Closer.create()) {
+            // https://github.com/google/error-prone/issues/2102
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java:303]: i'd put this comment on `@SuppressWarnings` line, because it matters there, not here

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveWriterFactory.java [hunk 9]
```diff
@@ -380,58 +380,53 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
...
-                schema = getHiveSchema(partition.get(), table);
+                    outputStorageFormat = partition.get().getStorage().getStorageFormat();
+                    schema = getHiveSchema(partition.get(), table);
 
-                writeInfo = locationService.getPartitionWriteInfo(locationHandle, partition, partitionName.get());
-            }
-            else if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.OVERWRITE) {
-                // Overwrite an existing partition
-                //
-                // The behavior of overwrite considered as if first dropping the partition and inserting a new partition, thus:
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveWriterFactory.java:419]: This comment is gone.

Please extract this commit to a separate PR.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java [hunk 10]
```diff
@@ -1179,10 +1179,12 @@ protected void doTestMismatchSchemaTable(
...
     protected void assertExpectedTableProperties(ConnectorTableProperties actualProperties, ConnectorTableProperties expectedProperties)
     {
         assertEquals(actualProperties.getPredicate(), expectedProperties.getPredicate());
         assertEquals(actualProperties.getDiscretePredicates().isPresent(), expectedProperties.getDiscretePredicates().isPresent());
+        // https://github.com/google/error-prone/issues/2101
```
Comment (findepi) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java:1187]: This should be at `@SuppressWarnin` line

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java (hunk 3 above)
Comment (wendigo) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java:298]: 2.5.1 was released with fixes for those false-positives so I'm updating this PR to drop suppressions

**Comment by wendigo** [comment]
@findepi @losipiuk error-prone 2.5.1 was released with fixes for false-positives reported by me. I've updated this PR so suppressions are no longer necessary.

File: core/trino-main/src/main/java/io/trino/execution/SqlStageExecution.java [hunk 11]
```diff
@@ -504,24 +504,26 @@ private synchronized void updateTaskStatus(TaskStatus taskStatus)
...
-            if (taskState == TaskState.FAILED) {
-                RuntimeException failure = taskStatus.getFailures().stream()
-                        .findFirst()
-                        .map(this::rewriteTransportFailure)
-                        .map(ExecutionFailureInfo::toException)
-                        .orElse(new TrinoException(GENERIC_INTERNAL_ERROR, "A task failed for an unknown reason"));
-                stateMachine.transitionToFailed(failure);
-            }
-            else if (taskState == TaskState.ABORTED) {
-                // A task should only be in the aborted state if the STAGE is done (ABORTED or FAILED)
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/execution/SqlStageExecution.java:516]: It seems the comment got  lost

**Review by findepi — CHANGES_REQUESTED** [review]
@wendigo please do split the PR. 

thanks for your work on this!

**Comment by wendigo** [comment]
AC @findepi. I've rebased and created a separate PR with the most controversial change: https://github.com/trinodb/trino/pull/6641

**Review by losipiuk — APPROVED** [review]

**Comment by losipiuk** [comment]
Thank you !
